File: scripts/run-sql-supabase.py
import os
import sys
import psycopg2

DB_PASSWORD = os.environ.get('DB_PASSWORD')
PROJECT_REF = 'nqcgnwpfxnbtdrvtkwej'
# IPv6 direct connect — db.nqcgnwpfxnbtdrvtkwej.supabase.co has no A record
DB_HOST = '2406:da14:271:9922:ba9b:5936:a55b:36a3'
DB_USER = 'postgres'
DB_NAME = 'postgres'
DB_PORT = 5432

SQL_PATH = os.path.join(os.path.dirname(__file__), '..', 'supabase', 'create-missing-tables.sql')

if not DB_PASSWORD:
    print('ERROR: Set DB_PASSWORD env var with your Supabase database password.')
    sys.exit(1)

with open(SQL_PATH, 'r', encoding='utf-8') as f:
    sql = f.read()

conn = None
try:
    print(f'Connecting to {DB_HOST}...')
    conn = psycopg2.connect(
        host=DB_HOST, port=DB_PORT, user=DB_USER,
        password=DB_PASSWORD, database=DB_NAME,
        sslmode='require'
    )
    conn.autocommit = True
    cur = conn.cursor()
    print('Connected. Executing SQL...')
    cur.execute(sql)
    print('SQL executed successfully.')

    # Verify tables exist
    cur.execute("""
        SELECT table_name FROM information_schema.tables
        WHERE table_schema = 'public'
          AND table_name IN ('product_features', 'deploy_history')
    """)
    rows = cur.fetchall()
    print('\nVerified tables:')
    for row in rows:
        print(f'  ✓ {row[0]}')
    if len(rows) < 2:
        print('  ⚠ Expected product_features and deploy_history — some tables may be missing.')

    cur.close()
except psycopg2.OperationalError as e:
    print(f'ERROR: {e}')
    if 'password authentication failed' in str(e):
        print('The DB_PASSWORD is incorrect.')
    sys.exit(1)
except Exception as e:
    print(f'ERROR: {e}')
    sys.exit(1)
finally:
    if conn:
        conn.close()

File: scripts/run-sql-supabase.mjs
// ============================================================
// Run SQL against Supabase Postgres via direct connection
// Usage: DB_PASSWORD="your-password" node scripts/run-sql-supabase.mjs
// ============================================================

import { readFileSync } from 'fs';
import { fileURLToPath } from 'url';
import { dirname, join } from 'path';
import pg from 'pg';

const { Client } = pg;
const __dirname = dirname(fileURLToPath(import.meta.url));

const DB_PASSWORD = process.env.DB_PASSWORD;
const PROJECT_REF = 'nqcgnwpfxnbtdrvtkwej';
const DB_HOST = `db.${PROJECT_REF}.supabase.co`;
const DB_USER = 'postgres';
const DB_NAME = 'postgres';
const DB_PORT = 5432;

async function main() {
  if (!DB_PASSWORD) {
    console.error('ERROR: Set DB_PASSWORD env var with your Supabase database password.');
    console.error('Get it from: Supabase Dashboard → Project Settings → Database → Connection String');
    process.exit(1);
  }

  const sqlPath = join(__dirname, '..', 'supabase', 'create-missing-tables.sql');
  const sql = readFileSync(sqlPath, 'utf-8');

  const client = new Client({
    host: DB_HOST,
    port: DB_PORT,
    user: DB_USER,
    password: DB_PASSWORD,
    database: DB_NAME,
    ssl: { rejectUnauthorized: false }
  });

  try {
    console.log(`Connecting to ${DB_HOST}...`);
    await client.connect();
    console.log('Connected. Executing SQL...');
    await client.query(sql);
    console.log('SQL executed successfully.');

    // Verify tables exist
    const res = await client.query(`
      SELECT table_name
      FROM information_schema.tables
      WHERE table_schema = 'public'
        AND table_name IN ('product_features', 'deploy_history')
    `);
    console.log('\nVerified tables:');
    for (const row of res.rows) {
      console.log(`  ✓ ${row.table_name}`);
    }
    if (res.rows.length < 2) {
      console.warn('  ⚠ Expected product_features and deploy_history — some tables may be missing.');
    }
  } catch (err) {
    console.error('ERROR:', err.message);
    if (err.message.includes('password authentication failed')) {
      console.error('The DB_PASSWORD is incorrect. Copy the password from your Supabase Connection String.');
    }
    if (err.message.includes('connect ETIMEDOUT') || err.message.includes('ECONNREFUSED')) {
      console.error('Cannot reach Supabase DB. Check your network or if IPv6 is required.');
    }
    process.exit(1);
  } finally {
    await client.end();
  }
}

main();
